### PR TITLE
Fix/no blocking on unconnected serial jtag

### LINKED ISSUE #31: Have a feature to choose `\r\n` or `\n` for new-lines

### LINKED ISSUE #32: `jtag_serial` blocks when USB is not connected
Would be nice to have this documented :)

### PR DESCRIPTION
**Pull request by bjoernQ**
Closes #31
Closes #32 

Additionally, this fixes the broken `no-op` feature

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -87,8 +124,16 @@ mod serial_jtag_printer {
                         }
                         conf.write_volatile(0b001);
 
+                        let mut timeout = TIMEOUT_ITERATIONS;
                         while conf.read_volatile() & 0b011 == 0b000 {
```
Comment (MabezDev) [inline src/lib.rs:128]: What about if we don't busy-wait here? If data isn't sent it should just overwrite the FIFO, right? I guess the question is, could we fill the FIFO and not send data in regular operation (i.e USB is connected).

**Comment by MabezDev** [comment]
I like the line ending / no-op changes, but I'm not so sure about the fix for #38. Currently, if you start the chip unconnected and connect via USB, it still won't print anything.

**Comment by bjoernQ** [comment]
> I like the line ending / no-op changes, but I'm not so sure about the fix for #38. Currently, if you start the chip unconnected and connect via USB, it still won't print anything.

I am also not super happy with this solution because of the case you mentioned. The only alternative I can think of is to have the timeout as low as possible and always trying to send data even if we will run into the timeout every time

File: src/lib.rs (hunk 1 above)
Comment (MabezDev) [reply src/lib.rs:128]: Yeah this isn't possible, we need to wait for the host to receive the packet.

**Comment by bjoernQ** [comment]
> I think what we can do is timeout initially, then on the next run check the `USB_SERIAL_JTAG_SERIAL_OUT_EP_DATA_AVAIL` bit, if this is set, it means data hasn't been sent to the host so you can bail early - not sure how it behaves when the device isn't connected though. Mind giving that a try?

Good idea! Will test that 👍

**Comment by MabezDev** [comment]
> > I think what we can do is timeout initially, then on the next run check the `USB_SERIAL_JTAG_SERIAL_OUT_EP_DATA_AVAIL` bit, if this is set, it means data hasn't been sent to the host so you can bail early - not sure how it behaves when the device isn't connected though. Mind giving that a try?
> 
> Good idea! Will test that +1

Another possible idea would be to check the USB address registers, but I don't know if they get cleared on disconnect - could be another thing to look into :)

**Comment by bjoernQ** [comment]
I think I now have a reliable working approach

From the TRM
> One thing to note is that there may be situations where the host is either not attached or the CDC-ACM virtual
port is not opened. In this case, the packets that are flushed to the host will never be picked up and the transmit
buffer will never be empty. It is important to detect this and time out, as this is the only way to reliably detect that
the port on the host side is closed.

So I guess this is the best option we have and it seems to work fine for me

**Review by MabezDev — APPROVED** [review]
LGTM, thanks for taking care of this!

**Pull request merged**